app/controllers/incomeSources/manage/ManageIncomeSourceController.scala

```diff
@@ -39,55 +41,53 @@ class ManageIncomeSourceController @Inject()(val manageIncomeSources: ManageInco
                                              val authorisedFunctions: AuthorisedFunctions,
                                              val retrieveNino: NinoPredicate,
                                              val retrieveIncomeSources: IncomeSourceDetailsPredicate,
-                                             val itvcErrorHandler: ItvcErrorHandler,
+                                             implicit val itvcErrorHandler: ItvcErrorHandler,
                                              implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
                                              val incomeSourceDetailsService: IncomeSourceDetailsService,
+                                             auditingService: AuditingService,
                                              val retrieveBtaNavBar: NavBarPredicate)
                                             (implicit val ec: ExecutionContext,
                                              implicit val sessionService: SessionService,
                                              implicit override val mcc: MessagesControllerComponents,
                                              val appConfig: FrontendAppConfig) extends ClientConfirmedController
   with FeatureSwitching with IncomeSourcesUtils {
 
-  def show(): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
-    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
-    implicit user =>
-      handleRequest(
-        sources = user.incomeSources,
-        isAgent = false,
-        backUrl = controllers.routes.HomeController.show().url
-      )
-  }
-
-  def showAgent(): Action[AnyContent] = Authenticated.async {
-    implicit request =>
-      implicit user =>
-        getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
-          implicit mtdItUser =>
-            handleRequest(
-              sources = mtdItUser.incomeSources,
-              isAgent = true,
-              backUrl = controllers.routes.HomeController.showAgent.url
-            )
-        }
+  def show(isAgent: Boolean): Action[AnyContent] = authenticatedAction(isAgent) { implicit user =>
+    handleRequest(
+      sources = user.incomeSources,
+      isAgent = isAgent,
+      backUrl = {
+        if(isAgent) controllers.routes.HomeController.show()
+        else controllers.routes.HomeController.showAgent
+      }.url
+    )
   }
 
   def handleRequest(sources: IncomeSourceDetailsModel, isAgent: Boolean, backUrl: String)
                    (implicit user: MtdItUser[_]): Future[Result] = {
-    if (isDisabled(IncomeSources)) {
-      Future.successful(Redirect(controllers.routes.HomeController.show()))
-    } else {
+
+    withIncomeSourcesFS {
       incomeSourceDetailsService.getViewIncomeSourceViewModel(sources) match {
         case Right(viewModel) =>
+          auditingService
```

Comment:
if we can move this code block between lines 81 and 82, this would reduce chances to have false audit record in case of Mongo failure to remove session data, essentially as I understand we audit in exact moment before rendering a view